crates/mpt/src/fetcher.rs

```diff
@@ -42,20 +46,23 @@ pub trait TrieDBFetcher {
     /// - Err(anyhow::Error): If the [Header] could not be fetched.
     ///
     /// [TrieDB]: crate::TrieDB
-    fn header_by_hash(&self, hash: B256) -> Result<Header>;
+    fn header_by_hash(&self, hash: B256) -> Result<Header, Self::Error>;
 }
 
-/// The [TrieDBHinter] trait defines the synchronous interface for hinting the host to fetch trie
+/// The [TrieHinter] trait defines the synchronous interface for hinting the host to fetch trie
 /// node preimages.
-pub trait TrieDBHinter {
+pub trait TrieHinter {
+    /// The error type for hinting trie node preimages.
+    type Error: Display + ToString;
```

Comment:
Do we actually need the error type here to be `Display + ToString`?

We could use the unit type for the `Noop` impls if not.